web/src/test/java/DtoValidationTest.java

```diff
@@ -0,0 +1,178 @@
+import edu.softserve.zoo.dto.BaseDto;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.stream;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Utility-Tester for matching DTOs (Data Transfer Objects) with their DOs (Domain Objects).
+ * Criteria for success includes such factors:
+ * <ul>
+ * <li>Matching number of fields: DTO has to have not more fields declared than the DO does.</li>
+ * <li>Matching naming of fields: DTO has to have same field naming as the DO does.</li>
+ * </ul>
+ * <p>
+ * Test created for using mapping models by convention.
+ * </p>
+ * This util stands not only for testing matching DTOs with their DOs,
+ * it also provides information about declared fields in tested Classes.
+ *
+ * @author Andrii Abramov on 13-May-16.
+ */
+public final class DtoValidationTest {
```

Comment:
apache commons has already implemented all the needed comparison stuff. Don't invent own bicycle. Use existing instead.